pkg/config/agentmanagement.go

```diff
@@ -96,12 +144,13 @@ type RemoteConfiguration struct {
 }
 
 type AgentManagementConfig struct {
-	Enabled         bool             `yaml:"-"` // Derived from enable-features=agent-management
-	Url             string           `yaml:"api_url"`
-	BasicAuth       config.BasicAuth `yaml:"basic_auth"`
-	Protocol        string           `yaml:"protocol"`
-	PollingInterval string           `yaml:"polling_interval"`
-	CacheLocation   string           `yaml:"remote_config_cache_location"`
+	Enabled                    bool             `yaml:"-"` // Derived from enable-features=agent-management
+	Url                        string           `yaml:"api_url"`
+	BasicAuth                  config.BasicAuth `yaml:"basic_auth"`
+	Protocol                   string           `yaml:"protocol"`
+	PollingInterval            string           `yaml:"polling_interval"`
+	CacheLocation              string           `yaml:"remote_config_cache_location"`
+	InvalidateCacheOnUrlChange bool             `yaml:"invalidate_cache_on_url_change"` // Whether to invalidate the cached config if the url hash is different
```

Comment:
I am not sure about this, probably we don't really need to make this configurable? (it makes sense in general not to fall back to something that is clearly a left over from a previous initial config)